pallets/sto/src/benchmarking.rs

```diff
@@ -0,0 +1,205 @@
+}
+
+fn setup_fundraiser<T: Trait>(complexity: u32, tiers: u32) -> Result<User<T>, DispatchError> {
+    let (alice, alice_portfolio) = user::<T>("alice");
+
+    create_assets_and_compliance::<T>(
+        &alice,
+        &[OFFERING_TICKER, RAISE_TICKER],
+        1_000_000,
+        complexity,
+    )?;
+
+    let venue_id = create_venue(&alice)?;
+
+    <Sto<T>>::create_fundraiser(
+        alice.origin().into(),
+        alice_portfolio,
+        OFFERING_TICKER,
+        alice_portfolio,
+        RAISE_TICKER,
+        generate_tiers::<T>(tiers),
+        venue_id,
+        None,
+        Some(101.into()),
+        0.into(),
+        vec![].into(),
+    )?;
+
+    Ok(alice)
+}
+
+fn user<T: Trait>(name: &'static str) -> (User<T>, PortfolioId) {
+    let user = <UserBuilder<T>>::default().generate_did().build(name);
+    let portfolio = PortfolioId::default_portfolio(user.did());
+    (user, portfolio)
+}
+
+benchmarks! {
+    _ {}
+
+    create_fundraiser {
+        // Number of tiers
+        let i in 1 .. MAX_TIERS as u32;
+
+        let (alice, alice_portfolio) = user::<T>("alice");
+
+         create_assets_and_compliance::<T>(&alice, &[OFFERING_TICKER, RAISE_TICKER], 1_000_000, 0)?;
+
+        let venue_id = create_venue(&alice)?;
+        let tiers = generate_tiers::<T>(i);
+    }: _(
+            alice.origin(),
+            alice_portfolio,
+            OFFERING_TICKER,
+            alice_portfolio,
+            RAISE_TICKER,
+            tiers,
+            venue_id,
+            None,
+            None,
+            0.into(),
+            vec![].into()
+        )
+    verify {
+        ensure!(FundraiserCount::get(OFFERING_TICKER) > 0, "create_fundraiser");
+    }
+
+    invest {
+        // Rule complexity
+        let c in 1 .. T::MaxConditionComplexity::get() as u32;
+
+        let  alice = setup_fundraiser::<T>(c, MAX_TIERS as u32)?;
+        let alice_portfolio = PortfolioId::default_portfolio(alice.did());
+        let (bob, bob_portfolio) = user::<T>("bob");
+
+        <Asset<T>>::unsafe_transfer(
+            alice_portfolio,
+            bob_portfolio,
+            &RAISE_TICKER,
+            1_000_000.into()
+        )?;
+    }: _(
+            bob.origin(),
+            bob_portfolio,
+            bob_portfolio,
+            OFFERING_TICKER,
+            0,
+            (MAX_TIERS as u128).into(),
+            Some(100.into()),
+            None
+        )
+    verify {
+        ensure!(<Asset<T>>::balance_of(&OFFERING_TICKER, bob.did()) > 0.into(), "invest");
+    }
+
+    freeze_fundraiser {
+        let alice = setup_fundraiser::<T>(0, 1)?;
+    }: _(alice.origin(), OFFERING_TICKER, 0)
+    verify {
+        ensure!(FundraiserCount::get(OFFERING_TICKER) > 0, "freeze_fundraiser");
+    }
+
+    unfreeze_fundraiser {
+        let alice = setup_fundraiser::<T>(0, 1)?;
+    }: _(alice.origin(), OFFERING_TICKER, 0)
+    verify {
+        ensure!(FundraiserCount::get(OFFERING_TICKER) > 0, "unfreeze_fundraiser");
```

Comment:
You need to first freeze the token then call unfreeze and verify that the fundraiser status is unfrozen (live)